lte/gateway/fabfile.py

```diff
@@ -484,6 +484,72 @@ def integ_test(
         env.hosts = [gateway_host]
 
 
+def integ_test_deb_installation(
+    gateway_host=None, test_host=None, trf_host=None,
+    destroy_vm='True', provision_vm='True',
+):
+    """
+    Run the integration tests. This defaults to running on local vagrant
+    machines, but can also be pointed to an arbitrary host (e.g. amazon) by
+    passing "address:port" as arguments
+
+    gateway_host: The ssh address string of the machine to run the gateway
+        services on. Formatted as "host:port". If not specified, defaults to
+        the `magma` vagrant box.
+
+    test_host: The ssh address string of the machine to run the tests on
+        on. Formatted as "host:port". If not specified, defaults to the
+        `magma_test` vagrant box.
+
+    trf_host: The ssh address string of the machine to run the TrafficServer
+        on. Formatted as "host:port". If not specified, defaults to the
+        `magma_trfserver` vagrant box.
+    """
+
+    destroy_vm = bool(strtobool(destroy_vm))
+    provision_vm = bool(strtobool(provision_vm))
+
+    # Setup the gateway: use the provided gateway if given, else default to the
+    # vagrant machine
+    gateway_ip = '192.168.60.142'
+
+    if not gateway_host:
+        gateway_host = vagrant_setup(
+            'magma_deb', destroy_vm, force_provision=provision_vm,
+        )
+    else:
+        ansible_setup(gateway_host, "deb", "magma_deb.yml")
+        gateway_ip = gateway_host.split('@')[1].split(':')[0]
+    execute(_restart_gateway)
+
+    # Setup the trfserver: use the provided trfserver if given, else default to the
+    # vagrant machine
+    if not trf_host:
+        trf_host = vagrant_setup(
+            'magma_trfserver', destroy_vm, force_provision=provision_vm,
+        )
+    else:
+        ansible_setup(trf_host, "trfserver", "magma_trfserver.yml")
+    execute(_start_trfserver)
+
+    # Run the tests: use the provided test machine if given, else default to
+    # the vagrant machine
+    if not test_host:
+        test_host = vagrant_setup(
+            'magma_test', destroy_vm, force_provision=provision_vm,
+        )
+    else:
+        ansible_setup(test_host, "test", "magma_test.yml")
```

Comment:
We do not need to start this machine as well?

Reply:
The integ tests require a magma test instance.